cmd/micromdm/serve.go

```diff
@@ -144,12 +159,15 @@ func serve(args []string) error {
 		removeService = block.LoggingMiddleware(logger)(svc)
 	}
 
-	devDB, err := devicebuiltin.NewDB(sm.DB)
-	if err != nil {
-		stdlog.Fatal(err)
-	}
+	//devDB, err := devicebuiltin.NewDB(sm.DB)
+	//if err != nil {
+	//	stdlog.Fatal(err)
+	//}
+	
+	devMysqlDB, err := devicemysql.NewDB(sm.MysqlDB)
 
-	devWorker := device.NewWorker(devDB, sm.PubClient, logger)
+	//devWorker := device.NewWorker(devDB, sm.PubClient, logger)
+	devWorker := device.NewWorker(devMysqlDB, sm.PubClient, logger)
```

Comment:
It would be best to not leave commented out lines in the code base. Additionally methods like `device.NewWorker` or `device.New` down below on line 237 take interfaces as input parameters for the DB argument. So you don't necessarily have to rename the `devDB` variable above to `devMysqlDB`. As long as your `devicemysql.NewDB` call returns an object that conforms to the interface needed you should be able to just say

```
devDB, err := devicemysql.NewDB(sm.MysqlDB)
```

And then the rest of the code should just work.

It would be preferred I think though to keep boltDB as the default and have the config setting dynamically choose the correct DB implementation. Just a couple initial thoughts.

Reply:
@knightsc Thank you for this review, left both lines in place so a developer could easily choose, which lines to enable/disable.

Even better would be a dynamic path based on given input flags.

I will modify the code base these days, so it will take into account, whether mysql settings are provided and switch to mysql or whether not and stick to the former approach of a boltdb.